### PR TITLE
Implement IPC::Run to run C binary with timeout

### PR DESCRIPTION
**Pull request by Jmeigs1**
Run linter, refactor and add to PrimerDesign test, setup dist:zilla to package via cpan file

### CODE REVIEW

File: lib/MPD/PrimerDesign.pm [hunk 1]
```diff
@@ -79,18 +82,26 @@ sub SayMppCmd {
...
+  my $ret;
+
+  eval{
+    # If command line arguments are ever required they are added as items in the
+    # cmd array
+    my @cmd = ($self->MpdBinary);
+    $ret = run(\@cmd,'<',$tmpCmdPt->stringify,'>',$mpdOut->stringify,timeout($timeout));
+  };
+
+  unless ($ret) {
```
Comment (akotlar) [inline lib/MPD/PrimerDesign.pm:105]: I think I would prefer $ret == 0

File: lib/MPD.pm [hunk 2]
```diff
@@ -38,15 +38,16 @@ has TwoBitFile  => ( is => 'ro', isa => AbsFile, coerce => 1, required => 1, );
 has MpdBinary   => ( is => 'ro', isa => AbsFile, coerce => 1, required => 1, );
 has MpdIdx      => ( is => 'ro', isa => File,    coerce => 1, required => 1, );
 has dbSnpIdx    => ( is => 'ro', isa => File,    coerce => 1, required => 1, );
-has OutExt => ( is => 'ro', isa => 'Str', required => 1, );
-has OutDir => ( is => 'ro', isa => AbsPath, coerce => 1, required => 1, );
-has Debug    => ( is => 'ro', isa => 'Bool', default => 0 );
-has RunIsPcr => ( is => 'ro', isa => 'Bool', default => 1 );
-has Act      => ( is => 'ro', isa => 'Bool', default => 0 );
+has timeout     => ( is => 'ro', isa => 'Int',   required => 1, );
```
Comment (akotlar) [inline lib/MPD.pm]: In general, maybe easier to review if place new param at end.

I also think we shouldn't require this; lets provide a default value

**Comment by akotlar** [comment]
Looks good; at a high level I would prefer this to have been 2 PRs: formatting / linter changes, and then timeout / IPC

**Review by akotlar — CHANGES_REQUESTED** [review]
Comments

File: lib/MPD.pm (hunk 2 above)
Comment (Jmeigs1) [reply lib/MPD.pm]: Agreed.  I had started to implement a default within the function runmpp but putting here is much cleaner.  Should not be required regardless.

**Comment by Jmeigs1** [comment]
Linting as part of CI has been drilled into me.  Apologies for the clutter.

A linting PR is overkill I think but should have probably been a separate commit or pushed directly.  Reviewing with linting changes is a headache. Will keep in mind.

File: lib/MPD.pm [hunk 3]
```diff
@@ -249,21 +250,20 @@ sub PrintPrimerData {
   my $isPcrPt = $self->OutDir->child( sprintf( "%s.isPcr.txt", $OutExt ) );
   $p->WriteIsPcrFile( $isPcrPt->stringify );
 
-#  my $compressPath;
-#  if($self->compress) {
-#    $compressPath = $self->compressPath( $self->OutDir->child($OutExt) );
-#  }
+  #  my $compressPath;
```
Comment (akotlar) [inline lib/MPD.pm]: Lets just remove the commented out code

**Pull request closed without merging**